File: js/graphSpeed.js
import {Graph} from "./GraphAdjMatrix.js"
import {randint} from "./helpers.js"

const N = 2000
const g = new Graph(false)

console.log("Graph size: ", N);

const t0 = Date.now()

for(let i = 0; i < N; i++) {
	g.addNode(i)
}

const t1 = Date.now()
console.log("addNodes: ", t1 - t0);

for(let i = 0; i < N; i++) {
	for(let j = 0; j < N; j++) {
		if(randint(1000) < 10)
			g.addEdge(i, j, randint(10))
	}
}

const t2 = Date.now()
console.log("addEdges: ", t2 - t1);

for(let i = 0; i < N; i++) {
	g.getNeighbours(i)
}

const t3 = Date.now()
console.log("getNeighbours: ", t3 - t2);

File: python/graphSpeed.py
from GraphAdjMatrix import Graph
from helpers import randint
import time

N = 2000
g = Graph(False)

t0 = 1000 * time.time()

for i in range(N):
	g.addNode(i)

t1 = 1000 * time.time()

for i in range(N):
	for j in range(N):
		if randint(1000) < 10:
			g.addEdge(i, j, randint(10))

t2 = 1000 * time.time()

for i in range(N):
	g.getNeighbours(i)

t3 = 1000 * time.time()

print("Graph size:", N)
print("addNodes", round(t1 - t0))
print("addEdges", round(t2 - t1))
print("getNeighbours", round(t3 - t2))
